app/migrations.go

```diff
@@ -287,6 +287,29 @@ func (a *App) DoSystemConsoleRolesCreationMigration() {
 	}
 }
 
+func (a *App) doContentExtractionConfigDefaultTrueMigration() {
+	if !a.Config().FeatureFlags.FilesSearch {
+		return
+	}
+	// If the migration is already marked as completed, don't do it again.
+	if _, err := a.Srv().Store.System().GetByName(ContentExtractionConfigDefaultTrueMigrationKey); err == nil {
+		return
+	}
+
+	a.UpdateConfig(func(config *model.Config) {
+		config.FileSettings.ExtractContent = model.NewBool(true)
+	})
+
+	system := model.System{
+		Name:  ContentExtractionConfigDefaultTrueMigrationKey,
+		Value: "true",
+	}
+
+	if err := a.Srv().Store.System().Save(&system); err != nil {
+		mlog.Critical("Failed to mark content extraction config migration as completed.", mlog.Err(err))
```

Comment:
Gentle reminder to check our logging [principles](https://developers.mattermost.com/contribute/server/style-guide/#log-levels) before merging this change.